airbyte/cloud/workspaces.py

```diff
@@ -97,21 +113,56 @@ def _organization_info(self) -> dict[str, Any]:
             client_secret=self.client_secret,
         )
 
-    @property
-    def organization_id(self) -> str | None:
-        """The ID of the organization this workspace belongs to.
+    @overload
+    def get_organization(self) -> CloudOrganization: ...
 
-        This value is cached after the first lookup.
-        """
-        return self._organization_info.get("organizationId")
+    @overload
+    def get_organization(
+        self,
+        *,
+        raise_on_error: Literal[True],
+    ) -> CloudOrganization: ...
 
-    @property
-    def organization_name(self) -> str | None:
-        """The name of the organization this workspace belongs to.
+    @overload
+    def get_organization(
+        self,
+        *,
+        raise_on_error: Literal[False],
+    ) -> CloudOrganization | None: ...
+
+    def get_organization(
+        self,
+        *,
+        raise_on_error: bool = True,
+    ) -> CloudOrganization | None:
+        """Get the organization this workspace belongs to.
+
+        Fetching organization info requires ORGANIZATION_READER permissions on the organization,
+        which may not be available with workspace-scoped credentials.
 
-        This value is cached after the first lookup.
+        Args:
+            raise_on_error: If True (default), raises AirbyteError on permission or API errors.
+                If False, returns None instead of raising.
+
+        Returns:
+            CloudOrganization object with organization_id and organization_name,
+            or None if raise_on_error=False and an error occurred.
+
+        Raises:
+            AirbyteError: If raise_on_error=True and the organization info cannot be fetched
+                (e.g., due to insufficient permissions).
         """
-        return self._organization_info.get("organizationName")
+        try:
+            info = self._organization_info
+        except AirbyteError:
+            if raise_on_error:
+                raise
+            return None
+
+        return CloudOrganization(
+            organization_id=str(info.get("organizationId", "")),
+            organization_name=info.get("organizationName"),
+        )
```

Comment:
Let's assume missing an organization ID or organization Name is fatal and will trigger our behavior to either raise or suppress according to the caller preference. Then, we can assert that ID and Name are both non-zero lenght strings and not null.